iothub/service/src/Query/QueryResponse.cs

```diff
@@ -3,194 +3,79 @@
-        /// </code>
-        /// </example>
-        /// <remarks>
-        /// Like with a more typical implementation of IEnumerator, this function should be called once before checking
-        /// <see cref="Current"/>.
-        ///
-        /// This function is async because it may make a service request to fetch the next page of results if the current page
-        /// of results has been advanced through already. Note that this function will return True even if it is at the end
-        /// of a particular page of items as long as there is at least one more page to be fetched.
-        /// </remarks>
-        public async Task<bool> MoveNextAsync(QueryOptions queryOptions = default, CancellationToken cancellationToken = default)
+        public override Stream ContentStream 
         {
-            cancellationToken.ThrowIfCancellationRequested();
+            get =>  _httpResponse.Content.ReadAsStreamAsync().Result; 
+            set => throw new NotImplementedException(); //TODO who needs this?
+        }
+        public override string ClientRequestId 
+        { 
+            get => throw new NotImplementedException(); 
+            set => throw new NotImplementedException(); 
+        }
 
-            if (_items.MoveNext())
-            {
-                // Current page of results still had an item to return to the user.
-                Current = _items.Current;
-                Debug.Assert(_items.Current != null);
-                return true;
-            }
+        public override void Dispose()
+        {
+            _httpResponse?.Dispose();
+        }
 
-            if (ContinuationToken == null && queryOptions?.ContinuationToken == null)
-            {
-                // The query has no more pages of results to return and the last page has been
-                // completely exhausted.
-                return false;
-            }
+        protected override bool ContainsHeader(string name)
+        {
+            return _httpResponse.Headers.Contains(name);
+        }
 
-            // User's can pass in a continuation token themselves, but the default behavior
-            // is to use the continuation token saved by this class when it last retrieved a page.
-            var queryOptionsClone = new JobQueryOptions
-            {
-                ContinuationToken = queryOptions?.ContinuationToken ?? ContinuationToken,
-                PageSize = queryOptions?.PageSize ?? _defaultPageSize,
-                JobType = _jobType,
-                JobStatus = _jobStatus,
-            };
+        protected override IEnumerable<HttpHeader> EnumerateHeaders()
+        {
+            return _httpHeaders;
+        }
 
-            if (!string.IsNullOrEmpty(_originalQuery))
+        protected override bool TryGetHeader(string name, out string value)
+        {
+            IEnumerable<string> outVariableHeaders = new List<string>();
+            bool found = _httpResponse.Headers.TryGetValues(name, out outVariableHeaders);
```

Comment:
```suggestion
            value = _httpResponse.Headers.TryGetValues(name, out outVariableHeaders)
                ? outVariableHeaders.First()
                : string.Empty;
```

Reply:
I'm not sure what you are going for here, but the ```HttpHeadersExtensions.SafeGetValue``` call trivializes this so I'll go with it instead

Reply:
Yeah, I left this comment before I saw the opportunity to use a helper we already had.